managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -762,6 +764,9 @@ public void asyncReadEntriesOrWait(int maxEntries, long maxSizeBytes, ReadEntrie
         } else {
             OpReadEntry op = OpReadEntry.create(this, readPosition, numberOfEntriesToRead, callback,
                     ctx, maxPosition);
+            if (op.readPosition == null) {
+                return;
```

Comment:
Maybe we need to call the method `callback.readEntriesFailed` to make sure the caller gets a result.

Reply:
Neet not call the method here, if `OpReadEntry` init failed, must ensure `OpReadEntry->readEntriesFailed` will be called, `callback.readEntriesFailed` will be called then.